assets/src/edit-story/app/uploader/useUploader.js

```diff
@@ -99,17 +99,37 @@ function useUploader() {
 
       // TODO: Move this check to useUploadMedia?
       if (!isValidType(file) && !canTranscodeFile) {
+        let options;
+        if (allowedFileTypes.length > 1) {
+          options = sprintf(
+            /* translators: %1$s: is a comma separated list of allowed file types, %2$s is an allowed file type. */
+            __('%1$s, or %2$s', 'web-stories'),
+            allowedFileTypes.slice(0, allowedFileTypes.length - 1).join(
+              /* translators: delimiter used in a list */
+              __(', ', 'web-stories')
+            ),
+            allowedFileTypes.slice(allowedFileTypes.length - 1)
+          );
+        } else if (allowedFileTypes.length === 1) {
+          options = allowedFileTypes[0];
+        }
```

Comment:
Nice job 👍 

Lists like that are hard in i18n because languages have very different rules for that stuff.

If you search for `delimiter used in a list` in the whole code base you'll find many more places where we would need such a solution.

Extracting such a logic into its own function would be great, so we could use it in all these places. It would also need to account for the case where there are exactly 2 items in the list.

See the [`wp_sprintf_l()`](https://developer.wordpress.org/reference/functions/wp_sprintf_l/) PHP function for an example. Note the useful translator comments and how the translatable strings contains both `%1$s` and `%2$s`

Reply:
I'll whip something up!

Reply:
You maybe able to just copy and paste the php code from `wp_sprintf_l` and turn it into javascript. 

If a `sprintf_l` is created, then it should live in the [i18n](https://github.com/google/web-stories-wp/tree/main/packages/i18n) package.

Reply:
Not a huge fan of the `wp_sprintf_l` API tbh, especially as in WordPress it needs to be used via `wp_sprintf`. Also, please don't just copy code for copyright reasons :-)

But adding such a list function to the `i18n` package sounds like the right place 👍

Reply:
Just updated the code - let me know what y'all think about the functions I created. Happy to change it however you feel is best.